src/app/item-popup/WeaponDeepsightInfo.tsx

```diff
@@ -1,27 +1,63 @@
+import BungieImage from 'app/dim-ui/BungieImage';
+import { PressTip } from 'app/dim-ui/PressTip';
+import { t } from 'app/i18next-t';
 import { DimItem } from 'app/inventory/item-types';
+import { isHarmonizable } from 'app/inventory/store/deepsight';
+import { useD2Definitions } from 'app/manifest/selectors';
 import Objective from 'app/progress/Objective';
+import { DEEPSIGHT_HARMONIZER } from 'app/search/d2-known-values';
 import styles from './WeaponDeepsightInfo.m.scss';
-
 /**
  * A progress bar that shows a weapon's crafting pattern progress.
  */
 export function WeaponDeepsightInfo({ item }: { item: DimItem }) {
   const record = item.patternUnlockRecord;
   const relevantObjectives = record?.objectives.filter((o) => !o.complete);
 
+  const defs = useD2Definitions()!;
+  const harmonizerIcon = defs.InventoryItem.get(DEEPSIGHT_HARMONIZER)?.displayProperties.icon;
+
   if (!relevantObjectives?.length) {
     return null;
   }
 
+  const harmonizable = isHarmonizable(item);
+
   return (
     <div className={styles.deepsightProgress}>
       {relevantObjectives && relevantObjectives.length > 0 && (
-        <div className={styles.deepsightProgressSection}>
-          {relevantObjectives.map((objective) => (
-            <Objective key={objective.objectiveHash} objective={objective} showHidden />
-          ))}
-        </div>
+        <>
+          {harmonizable ? (
+            <PressTip
+              tooltip={harmonizableTooltipContent(item)}
```

Comment:
In this case you're just calling a function that returns some JSX. At the end of the day, that's what components are and it's not necessarily making much of a difference here, but proper Components are used with the `<HarmonizableTooltipContent />` syntax.

I do wonder whether this whole thing (including the PressTip, the BungieImage, and the `useD2Definitions` call) should be a component.